config/configauth/configauth.go

```diff
@@ -28,7 +29,8 @@ type Authentication struct {
 
 // GetServerAuthenticator attempts to select the appropriate auth.Server from the list of extensions,
 // based on the requested extension name. If an authenticator is not found, an error is returned.
-func (a Authentication) GetServerAuthenticator(extensions map[component.ID]component.Component) (auth.Server, error) {
+// Deprecated: [v0.97.0] Use GetServerAuthenticatorContext instead.
+func (a Authentication) GetServerAuthenticatorContext(_ context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
```

Comment:
You should keep the old method `GetServerAuthenticator` deprecated and introduce the new one `GetServerAuthenticatorContext` not deprecated

Reply:
Just worked on it now @dmitryax 
Hope its okay now?